cli/src/alluxio.org/cli/cmd/process/start.go

```diff
@@ -0,0 +1,54 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package process
+
+import (
+	"github.com/palantir/stacktrace"
+	"github.com/spf13/cobra"
+
+	"alluxio.org/cli/env"
+)
+
+var Start = &StartCommand{}
+
+type StartCommand struct {
+	AsyncStart      bool
+	SkipKillOnStart bool
+}
+
+func (c *StartCommand) ToCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "start",
+		Short: "Start the process",
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			p, err := ParseSelectedProcessType()
+			if err != nil {
+				return stacktrace.Propagate(err, "error determining process to stop")
+			}
+
+			if !c.SkipKillOnStart {
+				_ = p.Stop(&env.StopOpts{})
+			}
+			return p.Start(&env.StartOpts{
+				AsyncStart: c.AsyncStart,
+			})
+		},
+	}
+	cmd.Flags().BoolVarP(&c.SkipKillOnStart, "skipKillOnStart", "N", false, "Avoid killing previous running processes when starting")
+	cmd.Flags().BoolVarP(&c.AsyncStart, "startAsync", "a", false, "Asynchronously start processes without monitoring for start completion")
+
+	for _, p := range env.ProcessRegistry {
+		p.SetStartFlags(cmd)
```

Comment:
The current implementation creates a lot of flags for the `bin/cli process start/stop` function. I was thinking maybe we could have a `bin/cli process start --type {master | worker | job_master | job_worker}` command. 
This would reduce the total number of flags, make the other flags more prevalent so that users know they exist, and most importantly, would allow for more seamless extensibility in the future. The word `type` lends itself great to start "combinations" of processes. For example, in the future (and I'm imagining this as I type), we could have something like `bin/cli process start --type cluster` or `bin/cli process start --type masters` where expanded definitions covering more process types for easier cluster management can be easily defined and operated over. 
wdyt?

Reply:
`--type` is possible, but it doesn't solve the concern about mixing process specific flags. see not possible, see https://github.com/Alluxio/alluxio/pull/17561#issuecomment-1577384655